### ISSUE TITLE
Plotting Indicator synchronization with resampled data

### ISSUE THREAD

**Issue by dimitar-petrov** [mementum/backtrader#277]
Hello mementum,

Great work with the backtesting framework and congrats on the responsive community.

There is case where indicator plotting is breaking in matplotlib due to data resampling.
The first datetime in data.datetime.plot() is smaller than self.pinf.xreal[0] and off by one error is happening in the loop that is counting.

```
ipdb> l
    184             for data in strategy.datas:
    185                 if not data.plotinfo.plot:
    186                     continue
    187 
    188                 import ipdb; ipdb.set_trace()
--> 189                 self.pinf.xdata = self.pinf.x
    190                 if len(data) < self.pinf.xlen:
    191                     self.pinf.xdata = xdata = []
    192                     xreal = self.pinf.xreal
    193                     dts = data.datetime.plot()
    194                     for dt in (x for x in dts if dt0 <= x <= dt1):

ipdb> len(self.pinf.xdata)
23004
ipdb> len(data)
413
ipdb> c
> /home/dpetrov/develop/python/py-envs/py3env/lib/python3.6/site-packages/backtrader/plot/plot.py(199)plot()
    198                 import ipdb; ipdb.set_trace()
--> 199                 for ind in self.dplotsup[data]:
    200                     self.plotind(

ipdb> l
    194                     for dt in (x for x in dts if dt0 <= x <= dt1):
    195                         dtidx = bisect.bisect_left(xreal, dt)
    196                         xdata.append(dtidx)
    197 
    198                 import ipdb; ipdb.set_trace()
--> 199                 for ind in self.dplotsup[data]:
    200                     self.plotind(
    201                         data,
    202                         ind,
    203                         subinds=self.dplotsover[ind],
    204                         upinds=self.dplotsup[ind],

ipdb> len(self.pinf.xdata)
412
ipdb> data.num2date(dt0)
datetime.datetime(2003, 1, 2, 8, 32)
ipdb> data.num2date(dt1)
datetime.datetime(2003, 3, 27, 14, 59)
ipdb> data.num2date(dts[0])
datetime.datetime(2003, 1, 2, 9, 0)
ipdb> c
n> /home/dpetrov/develop/python/py-envs/py3env/lib/python3.6/site-packages/backtrader/plot/plot.py(422)plotind()
    421             import ipdb; ipdb.set_trace()
--> 422             plottedline = pltmethod(self.pinf.xdata, lplot, **plotkwargs)
    423             try:

ipdb> n
ValueError: x and y must have same first dimension
> /home/dpetrov/develop/python/py-envs/py3env/lib/python3.6/site-packages/backtrader/plot/plot.py(422)plotind()
    421             import ipdb; ipdb.set_trace()
--> 422             plottedline = pltmethod(self.pinf.xdata, lplot, **plotkwargs)
    423             try:

ipdb> len(self.pinf.xdata)
412
ipdb> len(lplot)
413
ipdb> 
```
Hopefully my explanation was understandable.
I can set a minimal example if needed.

**Stack Trace:**
```
Traceback (most recent call last):
  File "./stester.py", line 1173, in <module>
    btrun()
  File "./stester.py", line 203, in btrun
    cerebro.plot(**pkwargs)
  File "/home/dpetrov/develop/python/py-envs/py3env/lib/python3.6/site-packages/backtrader/cerebro.py", line 659, in plot
    start=start, end=end)
  File "/home/dpetrov/develop/python/py-envs/py3env/lib/python3.6/site-packages/backtrader/plot/plot.py", line 207, in plot
    self.plotdata(data, self.dplotsover[data])
  File "/home/dpetrov/develop/python/py-envs/py3env/lib/python3.6/site-packages/backtrader/plot/plot.py", line 671, in plotdata
    self.plotind(data, ind, subinds=self.dplotsover[ind], masterax=ax)
  File "/home/dpetrov/develop/python/py-envs/py3env/lib/python3.6/site-packages/backtrader/plot/plot.py", line 422, in plotind
    plottedline = pltmethod(self.pinf.xdata, lplot, **plotkwargs)
  File "/home/dpetrov/develop/python/py-envs/py3env/lib/python3.6/site-packages/matplotlib/__init__.py", line 1819, in inner
    return func(ax, *args, **kwargs)
  File "/home/dpetrov/develop/python/py-envs/py3env/lib/python3.6/site-packages/matplotlib/axes/_axes.py", line 1382, in plot
    for line in self._get_lines(*args, **kwargs):
  File "/home/dpetrov/develop/python/py-envs/py3env/lib/python3.6/site-packages/matplotlib/axes/_base.py", line 381, in _grab_next_args
    for seg in self._plot_args(remaining, kwargs):
  File "/home/dpetrov/develop/python/py-envs/py3env/lib/python3.6/site-packages/matplotlib/axes/_base.py", line 359, in _plot_args
    x, y = self._xy_from_xy(x, y)
  File "/home/dpetrov/develop/python/py-envs/py3env/lib/python3.6/site-packages/matplotlib/axes/_base.py", line 219, in _xy_from_xy
    raise ValueError("x and y must have same first dimension")
ValueError: x and y must have same first dimension

```

**Comment by mementum** [comment]
The introduction of partial plotting added a calculation which was capping the end of the plotting array. In commit f3bba12026521121d9fe4ab61603cde8d4033f08 this is corrected and your pull request #278 is deactivated (by simply having an `if False` in place)

Can you check if the correction to partial plotting also corrects the problem?

**Comment by dimitar-petrov** [comment]
Hello mementum,

Yes commit f3bba12026521121d9fe4ab61603cde8d4033f08 is correcting the plotting multiple data with different sample-rates.

But once we introduce numfigs parameter it is breaking again:

Here is a debugging session:

```
> /home/dpetrov/develop/python/py-envs/py3env/lib/python3.6/site-packages/backtrader/plot/plot.py(212)plot()
    211 
--> 212                 for ind in self.dplotsup[data]:
    213                     self.plotind(

ipdb> l 200
    195 
    196                 self.pinf.xdata = self.pinf.x
    197                 if len(data) < self.pinf.xlen:
    198                     self.pinf.xdata = xdata = []
    199                     xreal = self.pinf.xreal
    200                     dts = data.datetime.plot()
    201                     for dt in (x for x in dts if dt0 <= x <= dt1):
    202                         dtidx = bisect.bisect_left(xreal, dt)
    203                         xdata.append(dtidx)
    204 
    205                     import ipdb; ipdb.set_trace()
    206                     if False:

ipdb> b 434
Breakpoint 1 at /home/dpetrov/develop/python/py-envs/py3env/lib/python3.6/site-packages/backtrader/plot/plot.py:434
ipdb> args
self = <backtrader.plot.plot.Plot_OldSync object at 0x7f5472d738d0>
strategy = <9J1G81MNN3.JohnDoe_C object at 0x7f546ca1fba8>
figid = 0
numfigs = 3
iplot = True
start = 0
end = 31587
kwargs = {}
ipdb> len(strategy.data0)
31587
ipdb> len(strategy.data1)
567
ipdb> c
> /home/dpetrov/develop/python/py-envs/py3env/lib/python3.6/site-packages/backtrader/plot/plot.py(434)plotind()
    433             pltmethod = getattr(ax, lineplotinfo._get('_method', 'plot'))
1-> 434             plottedline = pltmethod(self.pinf.xdata, lplot, **plotkwargs)
    435             try:

ipdb> len(lplot)
567
ipdb> len(self.pinf.xdata)
189
ipdb> 
```
